platform/pensando/sonic-platform-modules-dpu/dpu/utils/dpu_db_util.py

```diff
@@ -0,0 +1,602 @@
+                    entry = db.get_all(db.STATE_DB, tk)
+                    name = tk.replace(prefix, "")
+                    cause = entry['cause'] if 'cause' in entry else ""
+                    time = entry['time'] if 'time' in entry else ""
+                    user = entry['user'] if 'user' in entry else ""
+                    comment = entry['comment'] if 'comment' in entry else ""
+                    reboot_causes.append(
+                        [name, cause, comment, time, user]
+                    )
+                return reboot_causes
+            else:
+                log_info("Reboot-cause history is not yet available in StateDB")
+                return []
+        except Exception as e:
+            log_err(f"Failed to fetch reboot cause list due to {e}")
+            return []
+
+    def update(self):
+        log_info("Start reboot cause list updating")
+        try:
+            stat = self._fetch_reboot_cause_list()
+            if stat == None:
+                raise Exception("Failed to fetch reboot cause list")
+            self._refresh_reboot_cause_db(stat)
+        except Exception as e:
+            log_err(f'failed to fetch reboot cause data due to {e}')
+        log_info("End reboot cause list updating")
+
+    def _refresh_reboot_cause_db(self, stat):
+        name = NOT_AVAILABLE
+        try:
+            if self.db == None:
+                return
+            self.delete_table_entries()
+            if len(stat) > 10:
+                stat = stat[:10]
+            for entry in stat:
+                table_name = f"{self.table_pattern}{entry[0]}"
+                self.db.hset(table_name, 'cause', entry[1])
+                self.db.hset(table_name, 'comment', entry[2])
+                self.db.hset(table_name, 'device', f"DPU{self.slot_id}")
+                self.db.hset(table_name, 'time', entry[3])
+                self.db.hset(table_name, 'user', entry[4])
+        except Exception as e:
+            log_err('Failed to update system health values for {} - {}'.format(name, repr(e)))
+
+class VersionUpdater():
+    def __init__(self, chassis, db):
+        self.db = db
+        self.chassis = chassis
+        self.slot_id = get_slot_id(self.chassis)
+        self.table = f'{VERSION_UPDATE_TABLE_NAME}|DPU{self.slot_id}|1'
+        try:
+            self.apiHelper = APIHelper()
+            self.dpu_docker_name = self.apiHelper.get_dpu_docker_container_name()
+        except:
+            log_err('Failed to get dpu docker name')
+
+    def delete_table_version_entries(self):
+        try:
+            if self.db:
+                sonic_version_info_table = self.db.keys(self.table)
+                if len(sonic_version_info_table) == 1:
+                    table = sonic_version_info_table[0]
+                    self.db.delete(table)
+        except Exception as e:
+            log_err(f'failed to delete fetch sonic version entries due to {e}')
+
+    def _fetch_version_info(self):
+        try:
```

Comment:
@shanshri why do we need this?

Reply:
These versions are needed to show all the DPU FW currently used and is consumed by the "fwutil show status" and  "fwutil show updates" so that the user knows the running versions and if any update is required.

Reply:
Part of fwutil show.